File: modulo_6_cmms/internal/infrastructure/cache/redis.go
package cache

import (
    "context"
    "log"
    "os"

    "github.com/go-redis/redis/v8"
)

var ctx = context.Background()

func NewRedisClient() *redis.Client {
    redisURL := os.Getenv("REDIS_URL")
    if redisURL == "" {
        redisURL = "redis://localhost:6379"
    }

    opt, err := redis.ParseURL(redisURL)
    if err != nil {
        log.Printf("⚠️ Error parsing Redis URL: %v, usando localhost", err)
        opt = &redis.Options{
            Addr: "localhost:6379",
        }
    }

    client := redis.NewClient(opt)

    _, err = client.Ping(ctx).Result()
    if err != nil {
        log.Printf("⚠️ Error connecting to Redis: %v", err)
        return nil
    }

    log.Println("✅ Redis Client Connected (M6)")
    return client
}

File: modulo_14_1_transfer_window/src/config/redis.ts
import { createClient } from 'redis';

const redisClient = createClient({
    url: process.env.REDIS_URL || 'redis://localhost:6379'
});

redisClient.on('error', (err) => {
    console.error('❌ Redis connection error:', err);
});

redisClient.on('connect', () => {
    console.log('✅ Redis connected successfully (M14.1)');
});

redisClient.connect();

export default redisClient;
